include/h2o/memory.h

```diff
@@ -34,6 +34,18 @@ extern "C" {
 
 #define H2O_STRUCT_FROM_MEMBER(s, m, p) ((s *)((char *)(p)-offsetof(s, m)))
 
+#if defined(__clang__) || defined(__GNUC__) && (__GNUC__ > 2 || __GNUC__ == 2 && __GNUC_MINOR__ >= 5)
```

Comment:
Also, this line is super long, wasn't sure if it was ok to split up preprocessor statements across multiple lines.